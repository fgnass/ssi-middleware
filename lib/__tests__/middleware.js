const path = require('path');
const express = require('express');
const ssi = require('../middleware');

const request = require('request-promise-native');

let server;
let baseUrl;

const baseDir = path.resolve(__dirname, 'fixture');
const app = express();

beforeAll(
  () =>
    new Promise(resolve => {
      server = app.listen(0, () => {
        const { port } = server.address();
        baseUrl = `http://localhost:${port}`;

        app.use(
          ssi({
            baseDir,
            baseUrl
          })
        );
        app.get('/dynamic', (req, res) => {
          res.end('DYNAMIC CONTENT');
        });
        app.use(express.static(baseDir));

        resolve();
      });
    })
);

afterAll(() => {
  server.close();
});

test('middleware', () => {
  return request(`${baseUrl}`).then(body => {
    expect(body).toMatchSnapshot();
  });
});
