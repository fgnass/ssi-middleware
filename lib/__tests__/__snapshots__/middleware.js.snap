// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middleware 1`] = `
"<html>
  STATIC FILE

  DYNAMIC CONTENT
</html>
"
`;
